src/include/simeng/pipeline/PipelineBuffer.hh

```diff
@@ -29,6 +29,14 @@ class PipelineBuffer {
     headIsStart = !headIsStart;
   }
 
+  const T* getPendingSlots() const {
+    if (isStalled_) {
+      return getTailSlots();
+    } else {
+      return getHeadSlots();
+    }
+  }
```

Comment:
In relation to https://github.com/UoB-HPC/SimEng/pull/294/files#r1144819612, could an `isEmpty` function which loops through all buffer slots and checks for the presence of any instruction not be implemented? Something like this would also work if the buffer was expanded to more than a head and tail (i.e. 3 or 4+ buffer slots)

Reply:
I get the idea with an isEmpty function but am not sure it is so simple. As the function is templated it would be hard to define what constitutes a slot as empty e.g. in the in-order core there are ```pipelineBuffers``` containing ```Instructions``` as well as ```MacroOp```s. For the first, a slot is empty if it is a null pointer the other if the vector in a slot is of size 0. 

As for future proofing for a longer buffer; looking at the rest of the class it is very much tied to the idea of only having a length of 2 e.g. ```getTailSlots``` calculates a pointer based on the boolean value ```headIsStart```. I think an entire rework would be needed which is outside the scope of this PR.

Reply:
1. Given an initial value is passed to the constructor, this could be stored as a member variable and refered to by `isEmpty`. These values are `nullptr`, empty vector etc.
2. I still think that making this change now is worth while. Even if the rest of the class needed updating, this would be one area that wouldn't.

Reply:
This may be resolved by work done in the 32 bit support branch, which allows for changing the length of the buffer. I think we should leave as is in this branch. But implement what you are suggesting in the updated buffer